### PR TITLE
GFORMS-1888 - Add a feature to allow a choice to redirect a user to a…

### PR DESCRIPTION
**Pull request by DursunGundogan**
… url without opening a new tab

### CODE REVIEW

File: app/uk/gov/hmrc/gform/formtemplate/Rewriter.scala [hunk 1]
```diff
@@ -369,17 +377,29 @@ trait Rewriter {
         section match {
           case s: Section.NonRepeatingPage =>
             s.copy(
-              page = s.page.copy(includeIf = replaceIncludeIf(s.page.includeIf), fields = replaceFields(s.page.fields))
+              page = s.page.copy(
+                includeIf = replaceIncludeIf(s.page.includeIf),
+                fields = replaceFields(s.page.fields),
+                redirects = s.page.redirects.map(_.map(r => r.copy(ifExpr = replaceIncludeIf(r.ifExpr))))
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/formtemplate/Rewriter.scala:383]: Can we introduce `replaceRedirects` method and reuse it?

This
```
page.redirects.map(_.map(r => r.copy(ifExpr = replaceIncludeIf(r.ifExpr))))
```
Is repeated 3x here 👎

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/RedirectCtx.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  private val reads: Reads[RedirectCtx] =
+    ((JsPath \ "if")
+      .readNullable[IncludeIf] and
+      (JsPath \ "redirectUrl").read[String])(RedirectCtx.apply _)
+
+  private val writes: Writes[RedirectCtx] =
+    ((JsPath \ "if").writeNullable[IncludeIf] and
+      (JsPath \ "redirectUrl").write[String])(unlift(RedirectCtx.unapply))
+
+  implicit val format: Format[RedirectCtx] = Format[RedirectCtx](reads, writes)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/RedirectCtx.scala:37]: Same question as on frontend. What is the deal with nullability of `includeIf`?

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/RedirectCtx.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  private val reads: Reads[RedirectCtx] =
+    ((JsPath \ "if")
+      .read[IncludeIf] and
+      (JsPath \ "redirectUrl").read[String])(RedirectCtx.apply _)
+
+  private val writes: Writes[RedirectCtx] =
+    ((JsPath \ "if").write[IncludeIf] and
+      (JsPath \ "redirectUrl").write[String])(unlift(RedirectCtx.unapply))
+
+  implicit val format: Format[RedirectCtx] = Format[RedirectCtx](reads, writes)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/RedirectCtx.scala:37]: This cannot be derived?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/RedirectCtx.scala]: Yes, it can be derived. fixed

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**